### PR TITLE
Update output format of `osxcollector.output_filters.opendns.related_domains`

### PR DESCRIPTION
**Pull request by ivanlei**
A decent size rewrite of the filter and associated tests.
The filter is now more explicit about which domain in a line matched which initial domain.
A number of unicode handling bugs appear squashed

### CODE REVIEW

File: osxcollector/output_filters/opendns/related_domains.py [hunk 1]
```diff
@@ -10,14 +21,28 @@
...
+               'domain_in_line.com': ['related_domain.com'],
+               'another.com': ['1.2.3.4']
+           }
+        }
+    }
+    ```
     """
 
-    def __init__(self, initial_domains=None, initial_ips=None, generations=2, related_when=None):
+    def __init__(self, initial_domains=None, initial_ips=None, generations=DEFAULT_RELATED_DOMAINS_GENERATIONS, related_when=None):
```
Comment (jjsendor) [inline osxcollector/output_filters/opendns/related_domains.py]: Would be good to break the long line here as well.

Comment (ivanlei) [inline osxcollector/output_filters/opendns/related_domains.py]: Line length is declared and enforced to [140 chars in tox.ini](https://github.com/Yelp/osxcollector/blob/master/tox.ini#L19). This fits.

File: osxcollector/output_filters/opendns/related_domains.py [hunk 2]
```diff
@@ -69,35 +92,105 @@ def end_of_lines(self):
...
 
-        self._related_domains = filter(lambda x: not self._whitelist.match_values(x), list(self._related_domains))
-
-        for blob in self._all_blobs:
-            if self._related_domains and 'osxcollector_domains' in blob:
+        if domains_to_related:
+            for blob in self._all_blobs:
+                if 'osxcollector_domains' not in blob:
+                    continue
                 for domain in blob.get('osxcollector_domains'):
```
Comment (Qmando) [inline osxcollector/output_filters/opendns/related_domains.py]: minor:
You could replace 
```if 'osxcollector_domains' not in blob:
    continue```
by changing this line to 
```for domain in blob.get('osxcollector_domains', []):```

Comment (Qmando) [inline osxcollector/output_filters/opendns/related_domains.py]: minor:
You could replace 
```if 'osxcollector_domains' not in blob:
    continue```
by changing this line to 
```for domain in blob.get('osxcollector_domains', []):```

File: osxcollector/output_filters/opendns/related_domains.py [hunk 3]
```diff
@@ -69,35 +92,105 @@ def end_of_lines(self):
...
-                    if domain in self._related_domains:
+                    add_related_domains = False
+                    if domain in domains_to_related.keys():
```
Comment (Qmando) [inline osxcollector/output_filters/opendns/related_domains.py]: minor: 
```.keys()``` here is not necessary

**Pull request merged**